cleanup-passes/license-header.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import {EOL} from 'os';
+
+import * as promisify from 'promisify-node';
+import * as glob from 'glob';
+
+import {register} from '../cleanup-pass';
+import {ElementRepo} from '../element-repo.ts';
+import {existsSync, makeCommit} from './util';
+
+const polymerHeaderWithoutAtLicense =
+`Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt`;
+
+const polymerHeader =
+`@license
+${polymerHeaderWithoutAtLicense}`;
+
+const shebang = /^#!/;
+const doctype = /^<!\s*doctype/i;
+
+interface GlobOptions {
+  cwd: string;
+}
+
+const findFiles: (globPath: string, options: GlobOptions) => Promise<string[]> =
+    promisify(glob);
+
+function hasLicense(lines: string[], start: number, end: number): boolean {
+  for (let i = start; i < end; i++) {
+    if (lines[i].indexOf('@license') > -1) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function findSpliceLocation(lines: string[], regex: RegExp): number {
+  for (let i = 0; i < lines.length; i++) {
+    if (regex.test(lines[i])) {
+      return i;
+    }
+  }
+  return -1;
+}
+
+interface CommentLocation {
+  start: number;
+  end: number;
+}
+
+function findFirstComment(lines: string[], startToken: string, endToken: string): CommentLocation {
+  const o = {
+    start: -1,
+    end: -1
+  };
+  let i = 0;
+  for (i = 0; i < lines.length; i++) {
+    if (lines[i].indexOf(startToken) > -1) {
+      o.start = i;
+      break;
+    }
+  }
+  for (; i < lines.length; i++) {
+    if (lines[i].indexOf(endToken) > -1) {
+      o.end = i;
+      break;
+    }
+  }
+  return o;
```

Comment:
Optional: if the start and end tokens were regexps you could rewrite this whole function as:

```javascript
return {
  start: findSpliceLocation(lines, startToken),
  end: findSpliceLocation(lines, endToken)
}
```